tools/cloud-build/daily-tests/builds/gke-a3-high.yaml

```diff
@@ -0,0 +1,65 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+tags:
+- m.gke-cluster
+- m.gke-node-pool
+- m.vpc
+- m.multivpc
+- m.kubectl-apply
+- gke
+
+timeout: 14400s  # 4hr
+steps:
+- id: gke-a3-highgpu
+  name: us-central1-docker.pkg.dev/$PROJECT_ID/hpc-toolkit-repo/test-runner
+  entrypoint: /bin/bash
+  env:
+  - "ANSIBLE_HOST_KEY_CHECKING=false"
+  - "ANSIBLE_CONFIG=/workspace/tools/cloud-build/ansible.cfg"
+  args:
+  - -c
+  - |
+    set -x -e
+    cd /workspace && make
+    BUILD_ID_FULL=$BUILD_ID
+    BUILD_ID_SHORT=$${BUILD_ID_FULL:0:6}
+    EXAMPLE_BP=examples/gke-a3-highgpu.yaml
+
+    # Replacing the static subnet name to prevent collisions
+    sed  -i "s/gke-subnet-a3-high/gke-subnet-a3-high-$${BUILD_ID_SHORT}/" $${EXAMPLE_BP}
```

Comment:
Was this tested?  Looking at the a3high gke blueprint, it looks like this will match weirdly.  Notice that it's `gke-subnet-a3-highgpu` and you're missing the `gpu` part.  The `gpu` part was not in the mega blueprint, which is why this match is probably wrong.

Reply:
I had tested it out once before raising the CR. Let me share the test artifacts with you offline. I am ok making the change you have proposed. Can discuss it offline.

Reply:
I looked at the [a3-mega template code](https://github.com/GoogleCloudPlatform/cluster-toolkit/blob/develop/tools/cloud-build/daily-tests/builds/gke-a3-megagpu.yaml#L43). The high-gpu code follows the structure as defined here. If we are making changes in this template, should a change be made in the mega template as well? Let me know which convention we should follow. 

I made a commit with the gpu change. Testing with the gpu change was causing the pre-commit to fail as well. The integration test passes in both the cases.